agent/pkg/api/socket_routes.go

```diff
@@ -111,19 +111,35 @@ func websocketHandler(w http.ResponseWriter, r *http.Request, eventHandlers Even
 	}
 
 	for {
-		_, msg, err := ws.ReadMessage()
-		if err != nil {
-			if _, ok := err.(*websocket.CloseError); ok {
-				logger.Log.Debugf("Received websocket close message, socket id: %d", socketId)
-			} else {
-				logger.Log.Errorf("Error reading message, socket id: %d, error: %v", socketId, err)
+		// params[0]: query
+		// params[1]: enableFullEntries (empty: disable, non-empty: enable)
+		params := make([][]byte, 2)
+		breakWholeLoop := false
+		for i := range params {
+			_, params[i], err = ws.ReadMessage()
+			if err != nil {
+				if _, ok := err.(*websocket.CloseError); ok {
+					logger.Log.Debugf("Received websocket close message, socket id: %d", socketId)
+				} else {
+					logger.Log.Errorf("Error reading message, socket id: %d, error: %v", socketId, err)
+				}
+
+				breakWholeLoop = true
+				break
 			}
+		}
 
+		if breakWholeLoop {
```

Comment:
https://stackoverflow.com/questions/51996175/how-to-break-out-of-nested-loops-in-go

Reply:
@undera oh there are labels in Go right? Fixed it with https://github.com/up9inc/mizu/commit/7fadd84df85425f7e25e79b594142b92d8459dfc